lua/snippy.lua

```diff
@@ -144,23 +144,43 @@ end
 local function get_snippet_at_cursor()
     M.read_snippets()
     local _, col = unpack(api.nvim_win_get_cursor(0))
-    local current_line = api.nvim_get_current_line()
-    local word = current_line:sub(1, col):match('(%S+)$')
-    if word then
+
+    -- remove leading whitespace for current_line_to_col
+    local current_line_to_col =
+        api.nvim_get_current_line():sub(1, col):match('(%S+)$')
+
+    if current_line_to_col then
+        word = current_line_to_col:match('(%S+)$') -- remove leading whitespace
+        local word_bound = true
         local scopes = shared.get_scopes()
         while #word > 0 do
             for _, scope in ipairs(scopes) do
                 if scope and M.snippets[scope] then
                     if M.snippets[scope][word] then
-                        return word, M.snippets[scope][word]
+                        local snippet = M.snippets[scope][word]
+                        if snippet.modifier == "i" then
+                            -- match inside word
+                            return word, snippet
+                        elseif snippet.modifier == "b" then
+                            -- match if word is first on line
+                            if word == current_line_to_col then
+                                return word, snippet
+                            end
+                        elseif snippet.modifier == "w" or snippet.modifier == "" then
+                            if word_bound then
+                                -- by default only match on word boundary
+                                return word, snippet
+                            end
+                        else
+                            error( string.format(
+                                "Unkown modifier %s in snippet %s",
```

Comment:
```suggestion
                                "Unknown option %s in snippet %s",
```

I think UltiSnips call it 'options' instead of modifiers.